localstack-core/localstack/services/sns/v2/provider.py

```diff
@@ -644,15 +654,106 @@ def set_platform_application_attributes(
+        _attributes = {"Enabled": "true", "Token": token, **attributes}
+        if endpoint_arn and (endpoint := store.platform_endpoints.get(endpoint_arn)):
+            # endpoint for that application with that particular token already exists
+            if not endpoint["Attributes"] == _attributes:
+                raise InvalidParameterException(
+                    f"Invalid parameter: Token Reason: Endpoint {endpoint_arn} already exists with the same Token, but different attributes."
+                )
+            else:
+                return CreateEndpointResponse(EndpointArn=endpoint_arn)
+
+        endpoint_arn = create_platform_endpoint_arn(platform_application_arn)
+        endpoint = PlatformEndpoint(
+            Attributes=_attributes,
+            PlatformEndpointArn=endpoint_arn,
+            PlatformApplicationArn=platform_application_arn,
+        )
+        store.platform_endpoints[endpoint_arn] = endpoint
+        application["PlatformEndpoints"][token] = endpoint_arn
+
+        return CreateEndpointResponse(EndpointArn=endpoint_arn)
+
+    def delete_endpoint(self, context: RequestContext, endpoint_arn: String, **kwargs) -> None:
+        store = self.get_store(context.account_id, context.region)
+        endpoint = store.platform_endpoints.pop(endpoint_arn, None)
+        if endpoint:
+            platform_application = store.platform_applications.get(
+                endpoint["PlatformApplicationArn"]
+            )
+            if platform_application:
+                platform_application["PlatformEndpoints"].pop(endpoint["Attributes"]["Token"], None)
+
     def list_endpoints_by_platform_application(
         self,
         context: RequestContext,
         platform_application_arn: String,
         next_token: String | None = None,
         **kwargs,
     ) -> ListEndpointsByPlatformApplicationResponse:
-        # TODO: stub so cleanup fixture won't fail
-        return ListEndpointsByPlatformApplicationResponse(Endpoints=[])
+        store = self.get_store(context.account_id, context.region)
+        platform_application = store.platform_applications.get(platform_application_arn, {})
+        if not platform_application:
+            raise NotFoundException("PlatformApplication does not exist")
+        endpoint_arns = platform_application.get("PlatformEndpoints").values()
+        paginated_endpoint_arns = PaginatedList(endpoint_arns)
+        page, token = paginated_endpoint_arns.get_page(
+            token_generator=lambda x: get_next_page_token_from_arn(x),
+            page_size=100,
+            next_token=next_token,
+        )
+        response = ListEndpointsByPlatformApplicationResponse(
+            Endpoints=[
+                {
+                    "EndpointArn": endpoint_arn,
+                    "Attributes": store.platform_endpoints.get(endpoint_arn, {}).get("Attributes"),
+                }
+                for endpoint_arn in page
```

Comment:
this is slightly more complex than it needs to be due to the mentioned deviation of the type. A good example of the trade off I mentioned before